integration-tests/redis/src/main/java/io/quarkus/redis/it/QuarkusRedisResource.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.redis.it;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.lettuce.core.Value;
+
+@Path("/quarkus-redis")
+@ApplicationScoped
+public class QuarkusRedisResource {
+    @Inject
+    SyncCommands syncCommands;
+
+    @Inject
+    AsyncCommands asyncCommands;
+
+    @Inject
+    ReactiveCommands reactiveCommands;
+
+    // synchronous
+    @GET
+    @Path("/sync/{key}")
+    public String getSync(@PathParam("key") String key) {
+        return syncCommands.get(key);
+    }
+
+    @POST
+    @Path("/sync/{key}")
+    public void setSync(@PathParam("key") String key, String value) {
+        this.syncCommands.set(key, value);
+    }
+
+    @POST
+    @Path("/sync-bytes/{key}")
+    public void setSyncBytes(@PathParam("key") String key, String value) {
+        this.syncCommands.set(key, value.getBytes());
+    }
+
+    // hashes
+    @POST
+    @Path("/hashes/{key}")
+    public void hset(@PathParam("key") String key, String value) {
+        this.syncCommands.hset("hashes", key, value);
+    }
+
+    @GET
+    @Path("/hashes/")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Map<String, String> getHash() {
+        return this.syncCommands.hgetall("hashes");
+    }
+
+    // list
+    @POST
+    @Path("/lists/")
+    public void push(String value) {
+        this.syncCommands.lpush("strings", value);
+    }
+
+    @GET
+    @Path("/lists/")
+    public String pop() {
+        return this.syncCommands.lpop("strings");
+    }
+
+    // asynchronous
+    @GET
+    @Path("/async/{key}")
+    public void getAsync(@PathParam("key") String key, @Suspended final AsyncResponse asyncResponse) {
+        this.asyncCommands.get(key).thenAccept(value -> asyncResponse.resume(value));
```

Comment:
This should be using the new style of writing async jax-rs by returning a `CompletionStage<T>` instead of using the async response.

Reply:
Thanks, I'll try make this change.